src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3160,17 +3161,22 @@ static char* map_or_reserve_memory_aligned(size_t size, size_t alignment, int fi
     // Do manual alignment
     aligned_base = align_up(extra_base, alignment);
 
-    if (file_desc != -1) {
-      os::unmap_memory(extra_base, extra_size);
-    } else {
-      os::release_memory(extra_base, extra_size);
+    bool rc = (file_desc != -1) ? os::unmap_memory(extra_base, extra_size) :
+                                  os::release_memory(extra_base, extra_size);
+    assert(rc, "release failed");
+    if (!rc) {
+      aligned_base = NULL;
+      break;
```

Comment:
Seems like we can just `return NULL`  on this path, like we do with `extra_base == NULL` check in the same loop? So we can have:

```
if (!rc) {
  assert(false, "Unmap/release should always succeed here");
  return NULL;
}
```